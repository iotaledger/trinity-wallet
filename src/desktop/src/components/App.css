@import-normalize;

@custom-media --xs (min-width: 0px); /* default */
@custom-media --sm (max-width: 480px);
@custom-media --md (max-width: 768px);
@custom-media --lg (max-width: 1024px);
@custom-media --xl (max-width: 1440px);

:root {
    --color-text: white;
    --color-yellow: #f7d002;
    --color-blue: #8bd4ff;
    --color-red: #ff6c69;
    --color-green: #9dffaf;
    --color-darkgreen: green;

    --color-label: var(--color-yellow);
    --color-button-warning: var(--color-yellow);
    --color-button-danger: var(--color-red);
    --color-button-success: var(--color-green);
    --color-button-extra: var(--color-blue);
    --color-button-cta: var(--color-darkgreen);
    --color-button-default: white;

    /*--font-size-default: 1.8vh;*/
    --font-size-default: 14px;
    --font-size-small: 12px;
    --font-size-title: 20px;

    --z-index-modal: 2048;
    --z-index-overlay: 1024;
    --z-index-dropdown: 512;
    --z-index-elevated: 1;
}

* {
    &,
    &:before,
    &:after {
        box-sizing: border-box;
    }
}

html,
body {
    margin: 0;
    padding: 0;
    height: 100%;
}

body {
    background-image: url(../../../shared/images/bg-green.jpg);
    background-repeat: no-repeat;
    background-size: 100% 100%;
    background-attachment: fixed;
    color: var(--color-text);
    font-family: arial;
    font-size: var(--font-size-default);
    line-height: 1.5;
}

h1 {
    font-size: var(--font-size-title);
}

label {
    color: var(--color-label);
    display: block;
    margin-bottom: 8px;
}

select {
    &:focus {
        outline: 0;
    }
    background: rgba(0, 0, 0, 0.2);
    border: 0;
    padding: 16px 4px;
    width: 100%;
    border-bottom: 2px solid white;
    color: white;
}

option {
    background: transparent;
    color: black;
    padding: 4px;
}

input {
    box-sizing: border-box;

    &:focus {
        outline: none;
    }

    &[type='text'],
    &[type='email'],
    &[type='tel'],
    &[type='password'] {
        background: rgba(0, 0, 0, 0.2);
        border: 0;
        border-bottom: 2px solid white;
        color: white;
        padding: 16px;
        width: 100%;
    }
}

textarea {
    background: rgba(0, 0, 0, 0.2);
    border: 0;
    border-bottom: 2px solid white;
    box-sizing: border-box;
    color: white;
    padding: 16px;
    width: 100%;
    &:focus {
        outline: none;
    }
}
