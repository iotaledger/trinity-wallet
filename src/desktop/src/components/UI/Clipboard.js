import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { CopyToClipboard } from 'react-copy-to-clipboard';
import { showNotification } from 'actions/notifications';

import css from './Clipboard.css';

class Clipboard extends React.PureComponent {
    static propTypes = {
        text: PropTypes.string.isRequired,
        title: PropTypes.string.isRequired,
        success: PropTypes.string.isRequired,
        showNotification: PropTypes.func.isRequired,
    };

    render() {
        const { text, showNotification, title, success } = this.props;

        return (
            <CopyToClipboard text={text}>
                <span
                    className={css.clipboard}
                    onClick={() =>
                        showNotification({
                            type: 'success',
                            title: title,
                            text: success,
                        })
                    }
                >
                    {text}
                </span>
            </CopyToClipboard>
        );
    }
}

const mapStateToProps = () => ({});

const mapDispatchToProps = {
    showNotification,
};

export default connect(mapStateToProps, mapDispatchToProps)(Clipboard);
