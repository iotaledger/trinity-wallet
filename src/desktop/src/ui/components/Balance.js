import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { getSelectedSeed } from 'selectors/seeds';
import { formatValue, formatUnit, round } from 'libs/util';
import { getCurrencySymbol } from 'libs/currency';

class Balance extends React.PureComponent {
    static propTypes = {
        seed: PropTypes.object.isRequired,
        account: PropTypes.object.isRequired,
        settings: PropTypes.object.isRequired,
        marketData: PropTypes.object.isRequired,
    };

    render() {
        const { account, settings, marketData, seed } = this.props;
        const accountInfo = account.accountInfo[seed.name];

        if (!accountInfo) {
            return null;
        }

        const currencySymbol = getCurrencySymbol(settings.currency);
        const fiatBalance = round(
            accountInfo.balance * marketData.usdPrice / 1000000 * settings.conversionRate,
        ).toFixed(2);

        return (
            <div>
                <strong>{`${formatValue(accountInfo.balance)}${formatUnit(accountInfo.balance)}`}</strong>
                <small>{`${currencySymbol} ${fiatBalance}`}</small>
            </div>
        );
    }
}

const mapStateToProps = (state) => ({
    account: state.account,
    seed: getSelectedSeed(state),
    marketData: state.marketData,
    settings: state.settings,
});

export default connect(mapStateToProps)(Balance);
