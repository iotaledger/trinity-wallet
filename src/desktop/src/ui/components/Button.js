import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { Link } from 'react-router-dom';
import css from './button.scss';

/**
 * Button component
 */
export default class Button extends React.PureComponent {
    static propTypes = {
        /** Target link */
        to: PropTypes.string,
        /** Click event callback function */
        /** @param {object} event - Click event object */
        onClick: PropTypes.func,
        /** Button or link element content */
        children: PropTypes.node,
        /** Custom button style definitions */
        style: PropTypes.object,
        /** Button disabled state */
        disabled: PropTypes.bool,
        /** Button type */
        type: PropTypes.string,
        /** Buttons style type */
        variant: PropTypes.oneOf(['primary', 'secondary', 'positive', 'negative', 'extra', 'back']).isRequired,
        /** Buttons custom class */
        className: PropTypes.oneOf(['outline', 'small', 'large', 'square', 'inline', 'icon']),
        /** Buttons loading state */
        loading: PropTypes.bool,
    };

    static defaultProps = {
        variant: 'primary',
    };

    render() {
        const { onClick, children, className, to, variant, loading, style, type, disabled } = this.props;

        const loadingClass = loading ? css.loading : null;

        if (to) {
            return (
                <Link {...this.props} className={classNames(css.button, css[className], css[variant])}>
                    {children}
                </Link>
            );
        }

        return (
            <button
                style={style}
                type={type ? type : 'button'}
                onClick={onClick}
                disabled={disabled}
                className={classNames(css.button, css[className], css[variant], loadingClass)}
            >
                {children}
            </button>
        );
    }
}
