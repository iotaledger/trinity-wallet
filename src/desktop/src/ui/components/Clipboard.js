/* global Electron */
import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { generateAlert } from 'actions/alerts';

import css from './clipboard.scss';

/**
 * Copy to clipboard wrapper component
 */
class Clipboard extends React.PureComponent {
    static timeout = null;

    static propTypes = {
        /** Target content copied to clipboard */
        text: PropTypes.oneOfType([PropTypes.string, PropTypes.array]),
        /** Clipboard wrapper content */
        children: PropTypes.any,
        /** Timeout to clear the clipboard */
        timeout: PropTypes.number,
        /** Success notification title */
        title: PropTypes.string.isRequired,
        /** Success notification description */
        success: PropTypes.string.isRequired,
        /** @ignore */
        generateAlert: PropTypes.func.isRequired,
    };

    copy(e) {
        e.stopPropagation();

        const { text, generateAlert, title, success, timeout } = this.props;

        Electron.clipboard(text);
        generateAlert('success', title, success);

        if (timeout > 0) {
            if (this.timeout) {
                clearTimeout(this.timeout);
            }
            this.timeout = setTimeout(() => {
                Electron.clipboard('');
            }, timeout * 1000);
        }
    }

    render() {
        const { children, text } = this.props;

        return (
            <span className={css.clipboard} onClick={(e) => this.copy(e)}>
                {children || text}
            </span>
        );
    }
}

const mapDispatchToProps = {
    generateAlert,
};

export default connect(null, mapDispatchToProps)(Clipboard);
