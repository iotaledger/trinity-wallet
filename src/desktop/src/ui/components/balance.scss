.balance {
    height: 100%;
    display: flex;
    align-items: center;
    justify-content: center;
    flex-direction: column;

    @media (max-width: 320px) {
        height: 160px;
        padding-top: 15px;
        position: relative;
    }

    h3 {
        font-size: 14px;
        font-weight: 400;
        letter-spacing: 0.5px;
    }
    h1 {
        font-size: 64px;
        line-height: 64px;
        margin: 12px 0px;
        cursor: pointer;

        @media screen and (max-width: 860px) {
            margin: 4px 0;
        }

        small {
            font-size: 21px;
            line-height: 21px;
            margin-left: 6px;
        }
    }

    h2 {
        margin-bottom: 15px;
    }

    > a {
        position: absolute;
        top: 50%;
        left: 20px;
        margin-top: -5px;
        &:last-of-type {
            left: auto;
            right: 20px;
        }
    }
}
