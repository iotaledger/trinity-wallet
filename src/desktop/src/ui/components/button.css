@keyframes spin {
    to {
        transform: rotate(360deg);
    }
}

.button {
    position: relative;
    border: 1px solid transparent;
    border-radius: 3px;
    cursor: pointer;
    display: inline-block;
    height: 42px;
    line-height: 40px;
    font-size: var(--font-size-s);
    letter-spacing: 0.2px;
    min-width: 140px;
    text-decoration: none;
    text-transform: uppercase;
    text-align: center;
    font-weight: 200;
    padding: 0 24px 0;
    user-select: none;
    overflow: visible;
    transition: padding 0.2s cubic-bezier(0.55, 0, 0.1, 1);
    &:focus {
        outline: none;
    }
    &.small {
        padding: 0 16px;
        height: 32px;
        line-height: 32px;
        min-width: 30px;
        border-radius: 2px;
    }
    &.large,
    &.inline {
        padding: 0 32px;
        line-height: 44px;
        font-weight: 300;
        height: 46px;
        font-size: 16px;
        text-transform: capitalize;
    }
    &:disabled,
    &.disabled {
        opacity: 0.3;
        pointer-events: none;
    }
    &:before {
        content: '';
        box-sizing: border-box;
        position: absolute;
        top: 50%;
        left: 18px;
        width: 16px;
        height: 16px;
        margin: -8px 0 0 -8px;
        border-radius: 50%;
        border: 1px solid transparent;
        border-top-color: var(--color-Border);
        opacity: 0;
    }
}

footer {
    .button + .button {
        margin-left: 20px;
    }
}

.primary {
    composes: button;
    color: var(--primary-body);
    background: var(--primary);
    &:hover {
        background: var(--primary-hover);
    }
}

.secondary {
    composes: button;
    color: var(--secondary-body);
    background: var(--secondary);
    &:hover {
        background: var(--secondary-hover);
    }
}

.positive {
    composes: button;
    color: var(--positive-body);
    background: var(--positive);
    &:hover {
        background: var(--positive-hover);
    }
}

.negative {
    composes: button;
    color: var(--negative-body);
    background: var(--negative);
    &:hover {
        background: var(--negative-hover);
    }
}

.extra {
    composes: button;
    color: var(--extra-body);
    background: var(--extra);
    &:hover {
        background: var(--extra-hover);
        border-color: var(--extra-hover);
    }
}

.inline {
    background: none;
    &:hover {
        background: none;
    }
    &.secondary {
        color: var(--secondary);
        &:hover {
            color: var(--secondary-hover);
        }
    }
}

.outline {
    background: var(--body-bg);
    border: 1px solid;

    &:hover {
        background: var(--body-bg);
    }

    &.primary {
        color: var(--primary);
        &:hover {
            color: var(--primary-hover);
        }
    }

    &.secondary {
        color: var(--secondary);
        &:hover {
            color: var(--secondary-hover);
        }
    }
    &.positive {
        color: var(--positive);
        &:hover {
            color: var(--positive-hover);
        }
    }
    &.negative {
        color: var(--negative);
        &:hover {
            color: var(--negative-hover);
        }
    }
    &.extra {
        color: var(--extra);
        &:hover {
            color: var(--extra-hover);
        }
    }
}

.icon {
    border: none;
    background: none;
    text-transform: capitalize;
    font-size: 14px;
    color: var(--body);
    cursor: pointer;
    height: auto;
    padding: 0px;
    > span {
        display: block;
        cursor: pointer;
    }
    &:hover {
        background: none;
        > span {
            color: var(--primary);
        }
    }
}

.loading {
    pointer-events: none;
    padding: 0 12px 0 36px;
    &:before {
        transition: all 0.8s cubic-bezier(0.55, 0, 0.1, 1);
        opacity: 1;
        animation: spin 0.6s ease infinite;
    }
}
