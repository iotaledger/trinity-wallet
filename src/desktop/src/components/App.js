import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { persistStore } from 'redux-persist';
import { withRouter } from 'react-router-dom';
import store from 'store';
import i18next from 'libs/i18next';
import Loading from 'components/UI/Loading';
import Onboarding from 'components/Layout/Onboarding';
import Main from 'components/Layout/Main';
import Notifications from 'components/UI/Notifications';

import './App.css';

// should be used later:

// class ErrorBoundary extends React.Component {
//     static propTypes = {
//         children: PropTypes.node,
//     };
//
//     state = {};
//
//     componentDidCatch(error) {
//         this.setState(() => ({
//             error,
//         }));
//     }
//     render() {
//         if (this.state.error) {
//             return <p>{this.state.error.message}</p>;
//         }
//         return this.props.children;
//     }
// }

class App extends React.Component {
    static propTypes = {
        settings: PropTypes.shape({
            locale: PropTypes.string.isRequired,
            fullNode: PropTypes.string.isRequired,
        }).isRequired,
        app: PropTypes.shape({
            isOnboardingCompleted: PropTypes.bool.isRequired,
        }).isRequired,
    };

    state = {
        initialized: false,
    };

    componentWillMount() {
        persistStore(store, { blacklist: ['tempAccount', 'notifications', 'seeds'] }, () => {
            // setTimeout(
            //     () =>
            //         this.setState(() => ({
            //             initialized: true,
            //         })),
            //     2500,
            // );
            // TODO: re-add timeout to avoid flashes of the loading spinner. temporarily disabled for easier debugging
            this.setState(() => ({
                initialized: true,
            }));
        });
    }

    // TODO: this is not working on windows. Investigate why
    // componentDidMount() {
    //     const { remote } = require('electron');
    //     const keytar = remote.require('keytar');
    //     // const keytar = require('keytar');
    //     keytar
    //         .setPassword('iotaWallet', 'Main Wallet', 'AAABBBCCCC999')
    //         .then(() => {
    //             console.log('SAVED SEED');
    //         })
    //         .catch(err => {
    //             console.log('ERROR WHILE SAVING', err);
    //         });
    // }

    componentWillReceiveProps(nextProps) {
        if (nextProps.settings.locale !== this.props.settings.locale) {
            i18next.changeLanguage(nextProps.settings.locale);
        }
    }

    componentDidCatch(error) {
        this.setState(() => ({
            error,
        }));
    }

    render() {
        const { app } = this.props;

        if (this.state.initialized === false) {
            return <Loading />;
        }

        return (
            <div>
                {this.state.error && <p>{this.state.error.message}</p>}
                <Notifications />
                {app.isOnboardingCompleted ? <Main /> : <Onboarding />}
            </div>
        );
    }
}

const mapStateToProps = state => ({
    settings: state.settings,
    app: state.app,
});

export default withRouter(connect(mapStateToProps)(App));
