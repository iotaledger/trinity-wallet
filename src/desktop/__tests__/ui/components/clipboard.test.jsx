import React from 'react';
import { shallow } from 'enzyme';

import { ClipboardComponent as Clipboard } from 'ui/components/Clipboard';

const props = {
    text: 'Lorem ipsum',
    title: 'Foo',
    success: 'Bar',
    generateAlert: jest.fn(),
};

global.Electron = {
    clipboard: jest.fn(),
};

describe('Clipboard component', () => {
    test('Render the component', () => {
        const wrapper = shallow(<Clipboard {...props} />);

        expect(wrapper).toMatchSnapshot();
    });

    test('Text content', () => {
        const wrapper = shallow(<Clipboard {...props} />);

        expect(wrapper.text()).toEqual('Lorem ipsum');
    });

    test('Custom content', () => {
        const mockProps = Object.assign({}, props, { children: 'Fizz Buzz' });
        const wrapper = shallow(<Clipboard {...mockProps} />);

        expect(wrapper.text()).toEqual('Fizz Buzz');
    });

    test('Copy to clipboard event', () => {
        const wrapper = shallow(<Clipboard {...props} />);

        wrapper.simulate('click');

        expect(props.generateAlert).toHaveBeenCalledTimes(1);
        expect(Electron.clipboard).toHaveBeenCalledTimes(1);
    });
});
