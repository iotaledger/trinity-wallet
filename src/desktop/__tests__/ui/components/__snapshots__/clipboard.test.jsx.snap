// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Clipboard component Render the component 1`] = `
<span
  className="clipboard"
  onClick={[Function]}
>
  Lorem ipsum
</span>
`;
